art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
```

Comment:
I think there was a PR a few months ago that was never completed that exposed the Key in the Leaf Node.  (https://github.com/kellydunn/go-art/pull/8#issuecomment-151049436)  I think that waiting for that to merge and get finished up might be better before introducing a new exported struct.  What do you think?  Would that be sufficient for your purposes?